### ISSUE TITLE
msys2-devel package missing dependencies

### ISSUE THREAD

**Issue by earnie** [Alexpux/MSYS2-packages#1366]
In my quest to build `pacman` I needed to install `msys2-devel` which is expected.  What wasn't expected was the need to install `patch`, `pkg-config` and `make` after I installed msys2-devel package.  My opinion is that these should be required for `msys2-devel` as without them you cannot really do development for MSYS2.

```bash
$ pacman -S msys2-devel

:: There are 7 members in group msys2-devel:
:: Repository msys
   1) binutils  2) cocom  3) gcc  4) gcc-fortran  5) msys2-runtime-devel
   6) msys2-w32api-headers  7) msys2-w32api-runtime
```
```bash
==> Starting prepare()...
/usr/local/src/pacman/pacman/PKGBUILD: line 94: patch: command not found
==> ERROR: A failure occurred in prepare().
    Aborting...

$ pacman -S patch
```
```bash
checking pkg-config is at least version 0.9.0... ./configure: line 14838: /usr/bin/pkg-config: No such file or directory
no
checking for libarchive >= 3.0.0... no
configure: error: *** libarchive >= 3.0.0 is needed to compile pacman!
==> ERROR: A failure occurred in build().
    Aborting...

$ pacman -S pkg-config
```
```bash
/usr/local/src/pacman/pacman/PKGBUILD: line 131: make: command not found
==> ERROR: A failure occurred in build().
    Aborting...

$ pacman -S make
```

**Comment by Alexpux** [comment]
What needed to build packages and how to do it is described in wiki https://github.com/msys2/msys2/wiki/Creating-packages

**Comment by Alexpux** [comment]
@earnie, are you Earnie Boyd the original MSYS developer?

**Comment by earnie** [comment]
@Alexpux, yes that is me.  I'm really impressed with what you've done with MSYS2 and think your use of pacman as the installer of choice was a good idea.

I had read through the Creating-packages page before I began.  My process was to download the source package for pacman, cd to the source package and execute makepkg.  You suggested that building an existing package would be a good learning tool and it has been.  My issues were the missing tools required to build one of the main required packages.  However, I did finally manage to create a new pacman-5.1.0-4-x86_64.pkg.tar.xz file with makepkg.  It just took a little trial and error to complete it.

I'm considering of being a little more help here.  I'm tired of mingw.org, only a handful of people seem interested in keeping it and I'm not in that camp any longer.

**Comment by Alexpux** [comment]
@earnie wiki describe that for developing any packages you need install ````base-devel```` group and if you want to develop msys2 packages you also need install ````msys2-devel```` package. This is common build dependencies for all packages so we dont want to write tons of dependencies packages in every PKGBUILD

**Comment by mingwandroid** [comment]
Great to have you on board @earnie.

**Comment by earnie** [comment]
Hmm... yes it does, and I had noted it but forgot about it later.

I'm trying to think of why base-devel shouldn't be a required dependency of every other -devel package.

**Comment by Alexpux** [comment]
@earnie you can't specify GROUP in dependencies, only concrete package
